app/ui/browser-modern/views/browser/tabbar/tab-visuals.jsx

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import React, { Component } from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
```

Comment:
Some comments here would be helpful, I can't tell what the purpose of this is from the name/styling and how it's used in tab.jsx